File: BFS_DFS/bfs.cpp
#include <iostream>
#include <queue>
#include <vector>

using namespace std;

bool visited[9];
vector<int> graph[9];

void bfs(int start) {
    visited[start] = true;
    queue<int> q;
    q.push(start);
    while (!q.empty()) {
        int target = q.front();
        q.pop();
        cout << target << " ";
        for (int i = 0; i < graph[target].size(); ++i) {
            int node = graph[target][i];
            if (!visited[node]) {
                q.push(node);
                visited[node] = true;
            }
        }
    }
}

int main(void) {
    graph[1].push_back(2);
    graph[1].push_back(3);
    graph[1].push_back(8);
    graph[2].push_back(1);
    graph[2].push_back(7);
    graph[3].push_back(1);
    graph[3].push_back(4);
    graph[3].push_back(5);
    graph[4].push_back(3);
    graph[4].push_back(5);
    graph[5].push_back(3);
    graph[5].push_back(4);
    graph[6].push_back(7);
    graph[7].push_back(2);
    graph[7].push_back(6);
    graph[7].push_back(7);
    graph[8].push_back(1);
    graph[8].push_back(7);

    for (int i = 0; i < 9; ++i) {
        visited[i] = false;
    }
    bfs(1);
}

File: BFS_DFS/bfs.py
from collections import deque

def bfs(graph, x, visited):
    queue = deque([x])

    visited[x] = True

    while queue:
        target = queue.popleft()
        print(target, end=' ')
        for i in graph[target]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True


graph = [
    [],
    [2, 3, 8],
    [1, 7],
    [1, 4, 5],
    [3, 5],
    [3, 4],
    [7],
    [2, 6, 8],
    [1, 7]
]

visited = [False] * 9

bfs(graph, 1, visited)
